### PR TITLE
Add ref seq equals

### LINKED ISSUE #2063: Implement __eq__ for ReferenceSequence
We've no equality operator for reference sequences, so identical ones aren't equal in python:
```
>>> rs2 = rts.tables.reference_sequence
>>> rs1 = rts.tables.reference_sequence
>>> rs1 == rs2
False
```
... should they be?

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #2063

### CODE REVIEW

**Comment by benjeffery** [comment]
> What do we do with the relationship between `assert_equals` and `equals` elsewhere?

In almost all the others we are calling a low-level method in `equals`. Where we don't I think you're right that it makes sense to call `assert_equals`.

File: python/tests/test_reference_sequence.py [hunk 1]
```diff
@@ -164,26 +169,40 @@ def test_fails_attr_missing(self, ts_fixture, attr):
...
+    @pytest.mark.parametrize(
+        ("attr", "val"),
+        [
+            ("url", "foo"),
+            ("data", "bar"),
+            ("metadata", {"json": "runs the world"}),
+            ("metadata_schema", tskit.MetadataSchema(None)),
+        ],
+    )
+    def test_fails_different(self, ts_fixture, attr, val):
```
[the remark this answers is not in the document]
Comment (benjeffery) [reply python/tests/test_reference_sequence.py:188]: :+1:

File: python/tskit/metadata.py [hunk 2]
```diff
@@ -829,6 +829,12 @@ def metadata_bytes(self) -> Any:
     def nbytes(self) -> int:
         return len(self._ll_object.metadata) + len(self._ll_object.metadata_schema)
 
+    def equals(self, other):
```
[the remark this answers is not in the document]
Comment (benjeffery) [reply python/tskit/metadata.py:832]: :+1:

File: python/tskit/tables.py [hunk 3]
```diff
@@ -2773,7 +2773,18 @@ def asdict(self) -> dict:
             "url": self.url,
         }
 
-    def assert_equals(self, other, ignore_metadata=False):
+    def __eq__(self, other):
+        return self.equals(other)
+
+    def equals(self, other, ignore_metadata=False, ignore_url=False):
```
[the remark this answers is not in the document]
Comment (benjeffery) [reply python/tskit/tables.py:2779]: As it is similar to metadata a put it in, but yeah makes sense to wait till we work out what it is doing.

**Comment by benjeffery** [comment]
@jeromekelleher comments addressed! Thanks.

File: python/tskit/metadata.py [hunk 4]
```diff
@@ -829,6 +829,12 @@ def metadata_bytes(self) -> Any:
     def nbytes(self) -> int:
         return len(self._ll_object.metadata) + len(self._ll_object.metadata_schema)
 
+    def equals(self, other) -> bool:
```
Comment (jeromekelleher) [inline python/tskit/metadata.py:832]: Looks like we're not calling this any more -simplest to just delete it I guess?

**Review by jeromekelleher — APPROVED** [review]
LGTM! One thing to clear up before merging.

**Pull request merged**